art/attacks/evasion/boundary.py

```diff
@@ -315,7 +314,8 @@ def _orthogonal_perturb(self, delta, current_sample, original_sample):
             direction = np.swapaxes(direction, 0, self.classifier.channel_index - 1)
             for i in range(direction.shape[0]):
                 direction[i] /= np.linalg.norm(direction[i])
-                perturb[i] -= np.dot(perturb[i], direction[i]) * direction[i]
+                perturb[i] -= np.dot(np.dot(perturb[i], direction[i].T), direction[i])
```

Comment:
Is it correct that the changed line does the same as before and that the only change is that is uses explicitly numpy functions?

Reply:
Yes, this is confirmed with the notebooks' result of boundary attack, it works well with both untargeted and targeted attack on unsquared images.